Add OpenHarmony support to mach and CI

jschwe opened the pull request:
This allows building `servoshell` for OpenHarmony and tests this in CI.
Servoshell itself doesn't contain any OpenHarmony specific code yet, but 
still CI will help ensure that `libservo` and all dependencies compile for 
OpenHarmony. This is especially useful for gfx / fonts, since that area is
currently receiving some refactoring and CI for OpenHarmony would ensure
that the refactoring doesn't accidentally break the OpenHarmony build.

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors


- [x] There are tests for these changes: A CI job was added

jschwe committed:
Add ohos to mach

Signed-off-by: Jonathan Schwender <jonathan.schwender@huawei.com>

jschwe committed:
Add OpenHarmony build to CI

jschwe commented on .github/workflows/quick-check.yml:
This change is intended to be dropped after reviewal and before merging, and is just to demonstrate that the CI workflow works. The `try` run probably won't work until the change is merged to main.

sagudev replied:
You can just drop this and provide link to workflow run that was created via `mach try ohos`

jschwe replied:
Thanks! Dropped the commit and update the link.

jschwe commented:
Link to the ohos CI action run in my fork: https://github.com/jschwe/servo/actions/runs/9527245492

sagudev commented on .github/workflows/dispatch-workflow.yml:
```diff
@@ -57,3 +57,12 @@ jobs:
     with:
       profile: ${{ inputs.profile }}
+
+  OpenHarmony:
```
```suggestion
  ohos:
```
to be more consistent with the rest

sagudev commented:
There also need to be build-ohos in https://github.com/servo/servo/blob/main/.github/workflows/main.yml#L41 to actually run on mq.

sagudev reviewed (approved):
Changes to workflows and and try_parser looks good to me.

@mrobinson can you review the rest?

jschwe committed:
Rename ohos sdk action

I decided to rename the upstream ohos sdk action to
setup-ohos-sdk, making it clearer that is a github
action repository.

Signed-off-by: Jonathan Schwender <schwenderjonathan@gmail.com>

mukilan commented on .github/workflows/dispatch-workflow.yml:
```diff
@@ -57,3 +57,12 @@ jobs:
+    if: ${{ inputs.workflow == 'ohos' }}
+    name: OpenHarmony
+    uses: ./.github/workflows/ohos.yml
+    # I believe we don't need any secrets in the OpenHarmony CI workflow.
```
Perhaps we can remove the commented lines.

jschwe replied:
done

jschwe committed:
Remove commented line

Signed-off-by: Jonathan Schwender <jonathan.schwender@huawei.com>
